lms/djangoapps/lms_xblock/mixin.py

```diff
@@ -142,6 +144,62 @@ def _get_user_partition(self, user_partition_id):
 
         raise NoSuchUserPartitionError("could not find a UserPartition with ID [{}]".format(user_partition_id))
 
+    def get_parent_unit(self):
+        """
+        Finds xblock's parent unit if it exists.
+
+        Returns:
+            xblock: Returns the parent unit xblock if it exists.
+            If no parent unit exists, returns None.
+        """
+        xblock = self
+        while xblock:
+            xblock = xblock.get_parent()
```

Comment:
Agreed, though I would make the last line

xblock = parent

And I assume you can't just look for a vertical and then stop because verticals aren't always the direct child of a sequential.